app/validators/answers/number_answer_validator.py

```diff
@@ -71,6 +73,42 @@ def validate_value_in_limits(self):
                 self.MAXIMUM_GREATER_THAN_LIMIT, value=max_value, limit=MAX_NUMBER
             )
 
+        if isinstance(answer_source_max_value, dict):
+            source = answer_source_max_value.get("source")
+            answers_with_context = self.questionnaire_schema.numeric_answer_ranges
+            answer_source_max_value = (
+                self.questionnaire_schema.get_numeric_value_for_value_source(
+                    source, answer_source_max_value, answers_with_context
+                )
+            )
+            if (
+                isinstance(answer_source_max_value, int)
+                and answer_source_max_value > MAX_NUMBER
+            ):
+                self.add_error(
+                    self.MAXIMUM_GREATER_THAN_LIMIT,
+                    value=answer_source_max_value,
+                    limit=MAX_NUMBER,
+                )
+
+        if isinstance(answer_source_min_value, dict):
+            source = answer_source_min_value.get("source")
+            answers_with_context = self.questionnaire_schema.numeric_answer_ranges
+            answer_source_min_value = (
+                self.questionnaire_schema.get_numeric_value_for_value_source(
+                    source, answer_source_min_value, answers_with_context
+                )
+            )
+            if (
+                isinstance(answer_source_min_value, int)
+                and answer_source_min_value > MIN_NUMBER
```

Comment:
With negative numbers it should be `min < MIN_NUMBER`. `min > MIN_NUMBER` would mean if `-10` was the limit, `-11` would pass as `-11` is not greater than `-10`.

Reply:
Forgot to swap it when copying and pasting, updated 👍